### ISSUE TITLE
Tooltip rendering incorrectly on narrow viewports

### ISSUE THREAD

**Issue by bobwise** [sparkdesignsystem/spark-design-system#3806]
The Tooltip component is sometimes rendering incorrectly, causing the page to zoom unpredictably and requiring users to scroll to see all content.

![tooltip](https://user-images.githubusercontent.com/1424560/105874183-4de7c700-5fca-11eb-80e7-0a84b808d51d.gif)

## To Reproduce
1. Go to https://nifty-archimedes-e20fac.netlify.app/
2. Open Chrome Dev Tools and turn on Device Emulation (the icon next to the Elements tab)
3. Reload the page
4. Observe that the page can be scrolled horizontally
5. Open the Tooltip - the tooltip renders horizontally off the page
6. Opening and closing the tooltip repeatedly causes it to repeatedly recalculate its width and alter the viewport width 

## Expected Behavior
The Tooltip should be able to render in a viewport that is 320px or wider without requiring a horizontal scrollbar.

Reported in Angular but happening in all frameworks.

**Comment by afebbraro** [comment]
let's make sure to cover the bases with the window use for angular universal apps and tests,  if (typeof window !== 'undefined') {
